saleor/webhook/circuit_breaker/breaker_board.py

```diff
@@ -49,38 +59,79 @@ def validate_sync_events(self):
                     f'Event "{event}" is not supported by circuit breaker.'
                 )
 
-    def is_closed(self, app_id: int):
-        return self.storage.last_open(app_id) < (time.time() - self.cooldown_seconds)
-
-    def register_error(self, app_id: int):
-        errors = self.storage.register_event_returning_count(
-            app_id, "error", self.ttl_seconds
-        )
-        total = self.storage.register_event_returning_count(
-            app_id, "total", self.ttl_seconds
-        )
-
-        if total < self.failure_min_count:
-            return
-
-        if (errors / total) * 100 >= self.failure_threshold:
-            self.storage.update_open(app_id, int(time.time()))
-            logger.info(
-                "[App ID: %r] Circuit breaker tripped, cooldown is %r [seconds].",
-                app_id,
-                self.cooldown_seconds,
-            )
+    def update_breaker_state(self, app_id: int):
+        last_open, state = self.storage.last_open(app_id)
+        total = self.storage.get_event_count(app_id, "total") or 1
+        errors = self.storage.get_event_count(app_id, "error")
+        # CLOSED to OPEN
+        if state == CircuitBreakerState.CLOSED:
+            if (
+                total > self.failure_min_count
+                and (errors / total) * 100 >= self.failure_threshold
```

Comment:
Refactor this condition into method e.g. `if self.exceeded_error_threshold(total, errors):`